sound/soc/sof/intel/hda-loader-skl.c

```diff
@@ -418,10 +418,15 @@ static void cl_skl_cldma_fill_buffer(struct snd_sof_dev *sdev,
 	cl_skl_cldma_stream_run(sdev, true);
 }
 
-static int cl_skl_cldma_wait_interruptible(struct snd_sof_dev *sdev)
+static int cl_skl_cldma_wait_interruptible(struct snd_sof_dev *sdev, bool wait)
 {
 	int ret = 0;
 
+	if (!wait) {
+		usleep_range(3000, 5000);
```

Comment:
This makes no sense. if (!wait) then wait (by sleeping) ?

Reply:
There are 2 mode: interrupt inform and polling, so change to interrupt_wait should be better.

Reply:
My point is that callers to cl_skl_cldma_wait_interruptible() always set wait to false, hence you can hard code it to never wait for the IRQ.

Furthermore since IRQ is not used here for code loading, that flag that enables IRQ can be removed and hard coded to not enable the IRQ.